src/ucp/core/ucp_mm.c

```diff
@@ -175,11 +178,28 @@ static ucs_status_t ucp_mem_alloc(ucp_context_h context, size_t length,
 }
 
 
+static inline unsigned uct_flags(ucp_mem_map_params_t *params)
+{
+    unsigned flags = 0;
+
+    if (params && (params->field_mask & UCP_MEM_MAP_PARAM_FIELD_FLAGS)) {
+        if (params->flags & UCP_MEM_MAP_NONBLOCK) {
+            flags |= UCT_MD_MEM_FLAG_NONBLOCK;
+        }
+
+        if (params->flags & UCP_MEM_MAP_FIXED) {
+            flags |= UCT_MD_MEM_FLAG_FIXED;
+        }
+    }
+
+    return flags;
+}
+
 ucs_status_t ucp_mem_map(ucp_context_h context, ucp_mem_map_params_t *params,
                          ucp_mem_h *memh_p)
 {
     ucs_status_t status;
-    unsigned uct_flags;
+    unsigned flags = uct_flags(params);
```

Comment:
Please don't call the function `uct_flags()` here for several reasons:
1. it executes always despite of the fact the code below could avoid this execution
2. The return value might be wrong due to change in source data later.